### PR TITLE
Strong Number Using Indirect recursion

### PR DESCRIPTION
**Pull request by ErzaTitania-2001**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes
### Description

(Write your answer here.)

### Checklist

- [ yes] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Math Programs (WoC) #1527

### CODE REVIEW

**Comment by ErzaTitania-2001** [comment]
@iamrajiv  check my PR for any discrepancies and inform me if any! 
Issue in question Math Programs (WoC) #1527

File: Java/math/StrongRecursion.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//checking for a strong number using indirect recursion
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Add a description of the Algorithm/Code that you are looking to implement

File: Java/math/StrongRecursion.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.*;
+class StrongRecursion
+{
+    public static int sofd(int n) //recursive function for sum of factorial of digits
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Keep the Comments above the Line of Code not beside it

File: Java/math/StrongRecursion.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    {
+        if(n==0)
+            return n;
+        else
+            return (factorial(n%10)+sofd(n/10));
+
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Remove redundant lines of code

File: Java/math/StrongRecursion.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    {    
+        if (n == 0)    
+            return 1;    
+        else    
+            return(n * factorial(n-1));    
+    }    
+
+    public static void main()
+    {
+        Scanner sc=new Scanner(System.in);
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: You can try making the User Input more friendly

File: Java/math/StrongRecursion.java (hunk 2 above)
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Use meaningful function and variable names

File: Java/math/StrongRecursion.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ * N = 534
+ *  It is not a Strong Number
+ *  
+ *  
+ *  Time Complexity : O(n)
+ *  Space Complexity : 1
+ *  
+ * 
+ */
+
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Add a newline at the end of the file

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ErzaTitania-2001 — COMMENTED** [review]
Changes done @HarshCasper . Go over it.

File: Java/math/StrongRecursion.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Remove extra lines of code

File: Java/math/StrongRecursion.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        System.out.print(" Enter Number. N = ");
+        int num=sc.nextInt();
+        if(num==sumOfFactorialOfDigit(num)) 
+            System.out.println(" It is a Strong Number");
+        else
+            System.out.println(" It is not a Strong Number");
+    }
+}
+
+/* Sample Input And Output ;
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: `:` not `;`

File: Java/math/StrongRecursion.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+
+import java.util.*;
+class StrongRecursion
+{    
+    /*recursive function for sum of factorial of digits*/
+    public static int sumOfFactorialOfDigit(int n) 
+
+    {
+        if(n==0)
```
Comment (HarshCasper) [inline Java/math/StrongRecursion.java]: Format the Code with spaces around the Operators

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ErzaTitania-2001 — COMMENTED** [review]
If any other problems present than indentation, do say @HarshCasper

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@ErzaTitania-2001 Rename the program to `check_strong_number.java`. Also, Format your code using code formatter.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@ErzaTitania-2001 Rename the program `StrongRecursion.java` to `check_strong_number.java`.

**Review by iamrajiv — APPROVED** [review]

**Comment by ErzaTitania-2001** [comment]
@iamrajiv , should I close this pull request ? I want to pull another request on another program.

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Comment by ErzaTitania-2001** [comment]
@HarshCasper  @iamrajiv , I made a commit for the issue Bucket Sort #1590 in the same pull request  by mistake. Any solutions to this problem?

**Review by iamrajiv — APPROVED** [review]

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@ErzaTitania-2001 Remove the Bucket Sort program from this PR and make another PR for Bucket Sort. Also, whenever you make PR make separate branches for each PR rather than committing to the master branch.

**Comment by ErzaTitania-2001** [comment]
@iamrajiv Done. Deleted Bucket_Sort. When is the PR gonna be merged?

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**